spring5/src/test/java/org/jdbi/v3/spring5/TestJdbiFactoryBean.java

```diff
@@ -13,13 +13,137 @@
+
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.SqlStatements;
+import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.TestExecutionListeners;
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
 @RunWith(SpringJUnit4ClassRunner.class)
-@ContextConfiguration("/org/jdbi/v3/spring4/test-context.xml")
+@ContextConfiguration("/org/jdbi/v3/spring5/test-context.xml")
 @TestExecutionListeners(listeners = {DependencyInjectionTestExecutionListener.class})
-public class TestJdbiFactoryBean extends org.jdbi.v3.spring4.TestJdbiFactoryBean {}
+public class TestJdbiFactoryBean {
+
+    private Service service;
+    private DataSource ds;
+    private Jdbi jdbi;
+
+    @Autowired
+    public void setService(final Service service) {
+        this.service = service;
+    }
+
+    @Autowired
+    public void setDataSource(final DataSource dataSource) {
+        this.ds = dataSource;
+    }
+
+    @Autowired
+    public void setJdbi(final Jdbi jdbi) {
+        this.jdbi = jdbi;
+    }
+
+    @Test
+    public void testServiceIsActuallySet() {
+        assertThat(service).isNotNull();
+    }
+
+    @Test
+    public void testFailsViaException() {
+        assertThatExceptionOfType(ForceRollback.class).isThrownBy(() -> {
+            service.inPropagationRequired(aJdbi -> {
+                final Handle h = JdbiUtil.getHandle(aJdbi);
+                final int count = h.execute("insert into something (id, name) values (7, 'ignored')");
+                if (count == 1) {
+                    throw new ForceRollback();
+                } else {
+                    throw new RuntimeException("!ZABAK");
+                }
+            });
+        });
+
+        try (Handle h = Jdbi.open(ds)) {
+            final int count = h.createQuery("select count(*) from something").mapTo(int.class).one();
+            assertThat(count).isEqualTo(0);
+        }
+    }
+
+    @Test
+    public void testAutoInstalledPlugin() {
+        assertThat(jdbi.getConfig(SqlStatements.class).getAttribute(AutoPlugin.KEY))
+                .isEqualTo(AutoPlugin.VALUE);
+    }
+
+    @Test
+    public void testManualInstalledPlugin() {
+        assertThat(jdbi.getConfig(SqlStatements.class).getAttribute(ManualPlugin.KEY))
+                .isEqualTo(ManualPlugin.VALUE);
+    }
+
+    @Test
+    public void testGlobalDefinedAttribute() {
+        assertThat(jdbi.getConfig(SqlStatements.class).getAttribute("foo"))
+                .isEqualTo("bar"); // see test-context.xml
+    }
+
+    @Test
+    public void testNested() {
```

Comment:
github doesn't play well with these big diffs, I take it no meaningful content was changed?

Reply:
Only `spring4` -> `spring5` updates, no functional changes.